lib/Dialect/TritonGPU/Transforms/FP32DotTC.cpp

```diff
@@ -0,0 +1,93 @@
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "triton/Dialect/TritonGPU/Transforms/Passes.h"
+
+using namespace mlir;
+namespace tt = mlir::triton;
+
+namespace {
+
+// Implement 3xTF32 trick https://github.com/NVIDIA/cutlass/discussions/385
+// For a, b fp32
+// dot(a, b, f32Backend="3xtf32") ->
+//  let aBig = f32ToTF32(a), aSmall = a - aBig;
+//  let bBig = f32ToTF32(b), bSmall = b - bBig;
+//  dot(aSmall, bBig, f32Backend="tf32") +
+//  dot(aBig, bSmall, f32Backend="tf32") +
+//  dot(aBig, bBig, f32Backend="tf32")
+class TF32x3 : public OpRewritePattern<tt::DotOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult
+  matchAndRewrite(tt::DotOp dotOp,
+                  mlir::PatternRewriter &rewriter) const override {
+
+    auto isF32 = [](mlir::Value operand) {
+      return operand.getType()
+          .cast<RankedTensorType>()
+          .getElementType()
+          .isF32();
+    };
+
+    if (!(dotOp.getF32Backend() == "3xtf32" && isF32(dotOp.getOperand(0)) &&
+          isF32(dotOp.getOperand(1)))) {
+      return mlir::failure();
+    }
+
+    // Aux functions
+    auto f32ToTF32 = [&](mlir::Value value) -> mlir::Value {
+      return rewriter
+          .create<tt::ElementwiseInlineAsmOp>(
+              dotOp.getLoc(), value.getType(), "cvt.rna.tf32.f32 $0, $1;",
+              "=r,r",
+              /*isPure*/ true, /*pack*/ 1, std::vector<mlir::Value>{value})
```

Comment:
Nit, I don't think you need to construct a vector -- an ArrayRef or, if necessary SmallVector, should work?